robotcontainer.py

```diff
@@ -1,48 +1,259 @@
+            else:
+                self.angular_velocity_limit_ratio = (
+                    OP.angular_velocity_limit / OP.max_angular_velocity)
+
+        # Define a swerve drive subsystem by passing in a list of SwerveModules
+        # and some options
+        #
+        self.swerve = SwerveDrive(modules, navx_adapter, OP.max_speed, OP.max_angular_velocity)
+
+        # Set the swerve subsystem's default command to teleoperate using
+        # the controller joysticks
+        #
+        self.swerve.setDefaultCommand(
+            self.swerve.teleop_command(
+                translation=self.get_translation_input,
+                strafe=self.get_strafe_input,
+                rotation=self.get_rotation_input,
+                field_relative=SW.field_relative,
+                drive_open_loop=SW.drive_open_loop,
+            )
+        )
+
+    def log_data(self):
+        for pos in ("LF", "RF", "LB", "RB"):
+            encoder = getattr(self, f"{pos.lower()}_enc")
+            wpilib.SmartDashboard.putNumber(f"{pos} absolute encoder", encoder.absolute_position_degrees)
+            wpilib.SmartDashboard.putNumber(f"{pos} absolute encoder", encoder.absolute_position_degrees)
+
+    @staticmethod
+    def deadband(value, band):
+        return value if abs(value) > band else 0
+
+    def process_joystick_input(self, val, deadband=0.1, exponent=1, limit_ratio=1.0, invert=False):
+        """
+        Given a raw joystick reading, return the processed value after adjusting
+        for real-world UX considerations:
+          * apply a deadband to ignore jitter around zero
+          * apply an exponent for greater low-velocity control
+        """
+        deadbanded_input = self.deadband(val, deadband)
+        input_sign = +1 if val > 0 else -1  # this works for val=0 also
+        invert_sign = -1 if invert else +1
+        # abs required for fractional exponents
+        scaled_input = limit_ratio * abs(deadbanded_input) ** exponent
+        return invert_sign * input_sign * scaled_input
+
+    def get_translation_input(self, invert=True):
+        raw_stick_val = self.DriverController.getRawAxis(OP.translation_joystick_axis)
+        return self.process_joystick_input(raw_stick_val, invert=invert,
+                                           limit_ratio=self.speed_limit_ratio)
+
+    def get_strafe_input(self, invert=True):
+        raw_stick_val = self.DriverController.getRawAxis(OP.strafe_joystick_axis)
+        return self.process_joystick_input(raw_stick_val, invert=invert,
+                                           limit_ratio=self.speed_limit_ratio)
+
+    def get_rotation_input(self, invert=True):
+        raw_stick_val = self.DriverController.getRawAxis(OP.rotation_joystick_axis)
+        return self.process_joystick_input(
+            raw_stick_val, invert=invert, limit_ratio=self.angular_velocity_limit_ratio)
+
+    def get_autonomous_command(self):
```

Comment:
Creating this method means that now a `RobotContainer` has a method called `getAutonomousCommand()` and a method called `get_autonomous_command()`.  This will cause confusion.  We need to pick one and use it consistently.  (Of course, when we have multiple autonomous commands to choose from, we can use a `SendableChooser`.)